egui_extras/src/table.rs

```diff
@@ -352,24 +352,145 @@ pub struct TableBody<'a> {
+    ///
+    /// ### Example
+    /// ```
+    /// # egui::__run_test_ui(|ui| {
+    /// use egui_extras::{TableBuilderSize};
+    /// TableBuilder::new(ui)
+    ///     .column(Size::remainder().at_least(100.0))
+    ///     .body(|mut body| {
+    ///         let row_heights: Vec<f32> = vec![60.0, 18.0, 31.0, 240.0];
+    ///         body.heterogeneous_rows(row_heights.iter(), |row_index, mut row| {
+    ///             let thick = row_index % 6 == 0;
+    ///             row.col(|ui| {
+    ///                 ui.centered_and_justified(|ui| {
+    ///                     ui.label(row_index.to_string());
+    ///                 });
+    ///             });
+    ///             row.col(|ui| {
+    ///                 ui.centered_and_justified(|ui| {
+    ///                     ui.label(clock_emoji(row_index));
+    ///                 });
+    ///             });
+    ///             row.col(|ui| {
+    ///                 ui.centered_and_justified(|ui| {
+    ///                     ui.style_mut().wrap = Some(false);
+    ///                     if thick {
+    ///                         ui.heading("Extra thick row");
+    ///                     } else {
+    ///                         ui.label("Normal row");
+    ///                     }
+    ///                 });
+    ///             });
+    ///         });
+    ///     });
+    /// # });
+    /// ```
+    pub fn heterogeneous_rows(
+        &mut self,
+        heights: impl Iterator<Item = f32>,
+        mut populate_row: impl FnMut(usize, TableRow<'_, '_>),
+    ) {
+        // VIRTUAL_EXTENSION represents how far above and below the limits of the visible rectangle
+        // that we should consider when determining which rows to actually populate. this provides
+        // the illusion of table rows sliding out of view (rather than disappearing abruptly)
+        const VIRTUAL_EXTENSION: f32 = 30.0;
+
+        // in order for each row to retain its striped color as the table is scrolled, we need an
+        // iterator with the boolean built in based on the enumerated index of the iterator element
+        let mut striped_heights = heights
+            .enumerate()
+            .map(|(index, height)| (index % 2 == 0, height));
+
+        let max_height = self.end_y - self.start_y + VIRTUAL_EXTENSION;
+        let delta = self.delta() - VIRTUAL_EXTENSION;
+
+        // cumulative height of all rows above those being displayed
+        let mut height_above_visible: f64 = 0.0;
+        // cumulative height of all rows below those being displayed
+        let mut height_below_visible: f64 = 0.0;
+
+        let mut row_index = 0;
+
+        // calculate height above visible table range
+        while let Some((_, height)) = striped_heights.next() {
+            // when delta is greater than height above 0, we need to increment the row index and
+            // update the height above visble with the current height then continue
+            if height_above_visible >= delta as f64 {
```

Comment:
I don't quite follow the logic of everything, but it seems like this line should be taking `height` into account. If the new row height is so wide that it should be visible, then we should `break` and paint it. I think if you do that, then we can remove `VIRTUAL_EXTENSION` completely.

Reply:
While I think this suggestion is ultimately correct -- it solve the problem in the demo app where row index 0 isn't considered an "extra thick row" despite `0 % 6 == 0` (the logic that determines which rows should be thicker) -- it doesn't solve the same problem that I intended to solve with `VIRTUAL_EXTENSION`, which is that when scrolling up rows abruptly disappear rather than slide "under" the lip of the visible rectangle area. In theory even this `VIRTUAL_EXTENSION` constant may not work for sufficiently tall rows...